### PR TITLE
Document feed entry enclosures

### PR DESCRIPTION
**Pull request by chreekat**
Hopefully this will prevent the next lost soul from bleakly facing a new, mysterious 3-tuple

### CODE REVIEW

**Comment by snoyberg** [comment]
Pinging @nek0 for feedback. How do you feel about this change?

Note: this will be a major version bump, but I'm OK with doing so in this case.

**Comment by nek0** [comment]
Looks very good. It's more clear now.

**Comment by snoyberg** [comment]
@chreekat could you add some extra comments to this as well, including the `@since` annotations and a major version bump?

**Comment by chreekat** [comment]
Will do.

File: yesod-newsfeed/Yesod/FeedTypes.hs [hunk 1]
```diff
@@ -27,11 +28,27 @@ data Feed url = Feed
...
     { feedEntryLink    :: url
     , feedEntryUpdated :: UTCTime
     , feedEntryTitle   :: Text
     , feedEntryContent :: Html
-    , feedEntryEnclosure :: Maybe (url, Int, Text)
+    , feedEntryEnclosure :: Maybe (EntryEnclosure url)
+      -- ^ Allows enclosed data: RSS \<enclosure> or Atom \<link
+      -- rel=enclosure>
+      --
+      -- @since 1.5
```
Comment (chreekat) [inline yesod-newsfeed/Yesod/FeedTypes.hs]: The @\since here is for the field itself, not the particular type of the field.

**Comment by snoyberg** [comment]
Thanks!

**Pull request merged**